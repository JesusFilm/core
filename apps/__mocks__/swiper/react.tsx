import React from 'react'

export const Swiper = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperSlide = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperClass = {}
