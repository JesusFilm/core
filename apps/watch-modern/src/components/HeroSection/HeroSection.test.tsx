import { render, screen } from '@testing-library/react'

import { HeroSection } from './index'

describe('HeroSection', () => {
  it('renders correctly', () => {
    const { container } = render(<HeroSection />)
    expect(container).toMatchSnapshot()
  })

  it('displays the main heading', () => {
    render(<HeroSection />)
    const heading = screen.getByRole('heading', { level: 1 })
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('Watch the Greatest Story Ever Told')
  })

  it('displays the mission statement', () => {
    render(<HeroSection />)
    const missionText = screen.getByText(/Watch the life of Jesus through authentic films/)
    expect(missionText).toBeInTheDocument()
  })

  it('has a CTA button', () => {
    render(<HeroSection />)
    const ctaButton = screen.getByRole('button', { name: /Free Bible Videos/i })
    expect(ctaButton).toBeInTheDocument()
  })

  it('displays the language tagline', () => {
    render(<HeroSection />)
    const languageTagline = screen.getByText('ONE STORY. EVERY LANGUAGE.')
    expect(languageTagline).toBeInTheDocument()
  })

  it('displays the audience segmentation section', () => {
    render(<HeroSection />)
    const audienceHeading = screen.getByRole('heading', { level: 2 })
    expect(audienceHeading).toHaveTextContent('Start your journey today.')
  })

  it('has proper semantic structure', () => {
    render(<HeroSection />)
    const section = document.querySelector('section')
    expect(section).toBeInTheDocument()
  })

  it('has responsive padding to prevent header overlap', () => {
    render(<HeroSection />)
    const section = document.querySelector('section')
    expect(section).toHaveClass('pt-[60px]', 'sm:pt-[80px]', 'lg:pt-[120px]')
  })
}) 