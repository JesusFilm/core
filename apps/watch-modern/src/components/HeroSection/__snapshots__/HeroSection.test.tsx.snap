// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection renders correctly 1`] = `
<div>
  <section
    class="min-h-screen text-white relative flex items-end overflow-hidden pt-[60px] sm:pt-[80px] lg:pt-[120px]"
    style="background-color: rgb(12, 10, 9); background-blend-mode: normal;"
  >
    <div
      class="absolute inset-0 opacity-50 z-10"
      style="background-image: url(https://www.jesusfilm.org/_next/static/media/overlay.d86a559d.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
    />
    <div
      class="absolute inset-0 z-15"
    />
    <div
      class="absolute bottom-0 left-0 right-0 z-20 h-32"
    />
    <div
      class="max-w-[1920px] mx-auto px-4 sm:px-8 lg:px-20 pt-8 sm:pt-12 lg:pt-2 pb-12 sm:pb-16 lg:pb-24 relative z-20"
    >
      <div
        class="grid lg:grid-cols-2 gap-8 sm:gap-12 lg:gap-16 items-end"
      >
        <div
          class="max-w-5xl"
        >
          <div
            class="mb-4"
          >
            <div
              class="inline-flex items-center gap-2 mb-4"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-languages w-4 h-4 text-orange-400/80"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m5 8 6 6"
                />
                <path
                  d="m4 14 6-6 2-3"
                />
                <path
                  d="M2 5h12"
                />
                <path
                  d="M7 2h1"
                />
                <path
                  d="m22 22-5-10-5 10"
                />
                <path
                  d="M14 18h6"
                />
              </svg>
              <span
                class="text-orange-400/80 text-sm sm:text-lg tracking-wider"
              >
                ONE STORY. EVERY LANGUAGE.
              </span>
            </div>
          </div>
          <h1
            class="text-3xl sm:text-4xl md:text-6xl xl:text-7xl font-bold leading-[0.9] tracking-tight text-white mb-4 sm:mb-6"
          >
            Watch the 
            <span
              class="bg-gradient-to-r from-stone-200 to-stone-100 bg-clip-text text-transparent"
            >
              Greatest Story
            </span>
             Ever Told
          </h1>
          <p
            class="text-stone-100/90 text-base sm:text-xl md:text-2xl leading-relaxed max-w-3xl mb-6 sm:mb-8"
          >
            Watch the life of Jesus through authentic films, translated into thousands of languages and shared with billions of people worldwide.
          </p>
          <div
            class="flex flex-col sm:flex-row gap-4 items-start"
          >
            <button
              class="justify-center whitespace-nowrap ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-11 bg-white text-slate-900 hover:bg-blue-50 px-8 sm:px-12 py-4 sm:py-6 rounded-full font-semibold tracking-wide transition-all duration-200 flex items-center gap-3 shadow-2xl hover:shadow-white/20 text-base sm:text-lg"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-clapperboard sm:w-6 sm:h-6"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                style="width: 20px; height: 20px;"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.2 6 3 11l-.9-2.4c-.3-1.1.3-2.2 1.3-2.5l13.5-4c1.1-.3 2.2.3 2.5 1.3Z"
                />
                <path
                  d="m6.2 5.3 3.1 3.9"
                />
                <path
                  d="m12.4 3.4 3.1 4"
                />
                <path
                  d="M3 11h18v8a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2Z"
                />
              </svg>
              Free Bible Videos
            </button>
          </div>
        </div>
        <div
          class="lg:p-8"
        >
          <h2
            class="text-white text-2xl sm:text-4xl font-semibold mb-3 leading-tight"
          >
            Start your journey today.
          </h2>
          <p
            class="text-stone-200/80 text-base sm:text-xl leading-relaxed mb-6 sm:mb-8"
          >
            Find personalized videos and guidance based on your faith journey.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
